frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx

```diff
@@ -113,18 +115,21 @@ export const BackingStorage: React.FC<BackingStorageProps> = ({
   const [sc, scLoaded, scLoadError] = useK8sGet<ListKind<StorageClassResourceKind>>(
     StorageClassModel,
   );
+  const [infra, infraLoaded, infraLoadError] = useK8sGet<any>(InfrastructureModel, 'cluster');
   const isMCGStandalone = useFlag(GUARDED_FEATURES.ODF_MCG_STANDALONE);
 
   const formattedSS: StorageSystemSet = formatStorageSystemList(storageSystems);
-
   const hasOCS: boolean = formattedSS.has(STORAGE_CLUSTER_SYSTEM_KIND);
 
-  const allowedExternalStorage: ExternalStorage[] = SUPPORTED_EXTERNAL_STORAGE.filter(
-    ({ model }) => {
-      const kind = getStorageSystemKind(model);
-      return !formattedSS.has(kind);
-    },
-  );
+  const infraType = infra?.spec?.platformSpec?.type;
+  const enableRhcs = RHCS_SUPPORTED_INFRA.includes(infraType);
+
+  const allowedExternalStorage: ExternalStorage[] =
+    !enableRhcs || hasOCS
+      ? SUPPORTED_EXTERNAL_STORAGE.filter(
+          ({ model }) => getStorageSystemKind(model) !== STORAGE_CLUSTER_SYSTEM_KIND,
+        )
+      : SUPPORTED_EXTERNAL_STORAGE;
```

Comment:
dont show RHCS is one is created and platform is invalid.
This is legacy, we missed it.